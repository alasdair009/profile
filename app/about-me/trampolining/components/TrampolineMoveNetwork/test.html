<html lang="en">
  <head>
    <style>
      .videos {
        display: flex;
        height: 100vh;
        width: 100vw;
      }
      .videos div {
        align-items: center;
        display: flex;
        flex-direction: column;
        justify-content: center;
        width: 50%;
      }
      iframe {
        aspect-ratio: 16/9;
        width: 90%;
      }
    </style>
  </head>
  <body>
    <div class="videos">
      <div>
        <iframe
          id="standard"
          src=""
          data-src="https://www.youtube.com/embed/NpEaa2P7qZI?si=uJjK4C2TLWivF5BX"
          title="Standard Embed"
          allow="accelerometer; autoplay; clipboard-write; encrypted-media; gyroscope; picture-in-picture; web-share"
          referrerpolicy="strict-origin-when-cross-origin"
          allowfullscreen
        ></iframe>
        <button id="goStandard">Standard</button>
      </div>
      <div>
        <iframe
          id="privacy"
          src=""
          data-src="https://www.youtube-nocookie.com/embed/NpEaa2P7qZI?si=uJjK4C2TLWivF5BX"
          title="Enhanced privacy"
          allow="accelerometer; autoplay; clipboard-write; encrypted-media; gyroscope; picture-in-picture; web-share"
          referrerpolicy="strict-origin-when-cross-origin"
          allowfullscreen
        ></iframe>
        <button id="goPrivacy">Privacy</button>
      </div>
    </div>
    <script>
      const goStandard = document.getElementById("goStandard");
      const standardIFrame = document.getElementById("standard");
      const goPrivacy = document.getElementById("goPrivacy");
      const privacyIFrame = document.getElementById("privacy");
      goStandard.addEventListener(
        "click",
        function () {
          standardIFrame.setAttribute("src", standardIFrame.getAttribute("data-src"));
        },
        false
      );
      goPrivacy.addEventListener(
        "click",
        function () {
          privacyIFrame.setAttribute("src", privacyIFrame.getAttribute("data-src"));
        },
        false
      );
    </script>
  </body>
</html>
