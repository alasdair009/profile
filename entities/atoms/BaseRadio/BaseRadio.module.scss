@use "../../../app/styles/colors";
@use "../../../app/styles/dimensions";
@use "../../../app/styles/radii";
@use "../../../app/styles/helpers/rem";
@use "sass:map";

.root {
  appearance: none;
  aspect-ratio: 1;
  background: colors.$colorWhiteGhost;
  border: rem.toRem(1) solid colors.$colorblackEvil;
  border-radius: map.get(radii.$borderRadii, round);
  box-shadow: 0 0 rem.toRem(2) rem.toRem(3) colors.$colorblackEvil;
  width: rem.toRem(map.get(dimensions.$sizes, s32));

  &:checked {
    background: radial-gradient(
      colors.$colorGreenGrass rem.toRem(map.get(dimensions.$sizes, s8)),
      colors.$colorWhiteGhost rem.toRem(map.get(dimensions.$sizes, s8)),
      colors.$colorWhiteGhost
    );
  }

  &:invalid {
    border: rem.toRem(2) solid colors.$colorRedHeat;
  }
}
