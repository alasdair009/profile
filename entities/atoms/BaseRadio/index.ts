export { BaseRadio } from "./BaseRadio";
export type { BaseRadioProps } from "./BaseRadio";
