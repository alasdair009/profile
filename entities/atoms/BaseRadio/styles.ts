"use client";

import styled from "styled-components";
import { borderRadii, colors, sizes } from "@/entities";
import { rem } from "polished";

export const Root = styled.input<{ $isInvalid: boolean }>`
  appearance: none;
  aspect-ratio: 1;
  background: ${colors.whiteGhost};
  border: ${rem(1)} solid ${colors.blackEvil};
  border-radius: ${borderRadii.round};
  box-shadow: 0 0 ${rem(2)} ${rem(3)} ${colors.blackEvil};
  width: ${sizes.s32.rem};

  &:checked {
    background: radial-gradient(
      ${colors.greenGrass} ${sizes.s8.rem},
      ${colors.whiteGhost} ${sizes.s8.rem},
      ${colors.whiteGhost}
    );
  }
`;
