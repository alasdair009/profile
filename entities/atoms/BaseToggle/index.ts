export { BaseToggle } from "./BaseToggle";
export type { BaseToggleProps } from "./BaseToggle";
