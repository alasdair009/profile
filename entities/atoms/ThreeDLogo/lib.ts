import * as THREE from "three";
import { SVGLoader } from "three/addons/loaders/SVGLoader.js";

const fillMaterial = new THREE.MeshBasicMaterial({ color: "#F3FBFB" });
const stokeMaterial = new THREE.LineBasicMaterial({
  color: "#00A5E6",
});

const renderSVG = (extrusion, svg: string) => {
  const loader = new SVGLoader();
  const svgData = loader.parse(svg);
  const svgGroup = new THREE.Group();
  const updateMap = [];

  svgGroup.scale.y *= -1;
  svgData.paths.forEach((path) => {
    const shapes = SVGLoader.createShapes(path);

    shapes.forEach((shape) => {
      const meshGeometry = new THREE.ExtrudeGeometry(shape, {
        depth: extrusion,
        bevelEnabled: false,
      });
      const linesGeometry = new THREE.EdgesGeometry(meshGeometry);
      const mesh = new THREE.Mesh(meshGeometry, fillMaterial);
      const lines = new THREE.LineSegments(linesGeometry, stokeMaterial);

      updateMap.push({ shape, mesh, lines });
      svgGroup.add(mesh, lines);
    });
  });

  const box = new THREE.Box3().setFromObject(svgGroup);
  const size = box.getSize(new THREE.Vector3());
  const yOffset = size.y / -2;
  const xOffset = size.x / -2;

  // Offset all of group's elements, to center them
  svgGroup.children.forEach((item) => {
    item.position.x = xOffset;
    item.position.y = yOffset;
  });
  svgGroup.rotateX(-Math.PI / 2);

  return {
    object: svgGroup,
    update(extrusion) {
      updateMap.forEach((updateDetails) => {
        const meshGeometry = new THREE.ExtrudeGeometry(updateDetails.shape, {
          depth: extrusion,
          bevelEnabled: false,
        });
        const linesGeometry = new THREE.EdgesGeometry(meshGeometry);

        updateDetails.mesh.geometry.dispose();
        updateDetails.lines.geometry.dispose();
        updateDetails.mesh.geometry = meshGeometry;
        updateDetails.lines.geometry = linesGeometry;
      });
    },
  };
};

const setupScene = (container) => {
  const scene = new THREE.Scene();
  const renderer = new THREE.WebGLRenderer({ antialias: true, alpha: true });
  const camera = new THREE.PerspectiveCamera(
    50,
    window.innerWidth / window.innerHeight,
    0.01,
    1e5
  );
  const ambientLight = new THREE.AmbientLight("#888888");
  const pointLight = new THREE.PointLight("#ffffff", 2, 800);
  const controls = new OrbitControls(camera, renderer.domElement);
  const animate = () => {
    renderer.render(scene, camera);
    controls.update();

    requestAnimationFrame(animate);
  };

  renderer.setSize(window.innerWidth, window.innerHeight);
  scene.add(ambientLight, pointLight);
  camera.position.z = 50;
  camera.position.x = 50;
  camera.position.y = 50;
  controls.enablePan = false;

  container.append(renderer.domElement);
  window.addEventListener("resize", () => {
    camera.aspect = window.innerWidth / window.innerHeight;
    camera.updateProjectionMatrix();
    renderer.setSize(window.innerWidth, window.innerHeight);
  });
  animate();

  return scene;
};
