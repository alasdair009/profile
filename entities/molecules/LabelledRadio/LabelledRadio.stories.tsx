import { Meta, StoryObj } from "@storybook/react";
import { LabelledRadio } from "./LabelledRadio";
import { expect, within } from "@storybook/test";
import { BaseRadio } from "../../atoms/BaseRadio";

const meta: Meta<typeof LabelledRadio> = {
  component: LabelledRadio,
  args: {
    name: "radioentry",
    required: true,
    isInvalid: false,
    label: "Radio entry",
  },
};
export default meta;

export const Default: StoryObj<typeof LabelledRadio> = {
  play: async ({ canvasElement }) => {
    const canvas = within(canvasElement);
    const labelledRadio = canvas.getByTestId(LabelledRadio.name);

    await expect(labelledRadio).toBeInTheDocument();
  },
};

export const Checked: StoryObj<typeof LabelledRadio> = {
  args: {
    checked: true,
  },
  play: async ({ canvasElement }) => {
    const canvas = within(canvasElement);
    const labelledRadio = canvas.getByTestId(LabelledRadio.name);
    const baseRadioElement = canvas.getByTestId(BaseRadio.name);

    await expect(baseRadioElement).toHaveAttribute("checked");
  },
};
