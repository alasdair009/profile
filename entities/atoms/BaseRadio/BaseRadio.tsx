"use client";
import { Root } from "./styles";
import { HTMLAttributes, useState } from "react";

export type BaseRadioProps = {
  /**
   * If the radio should be displayed as invalid.
   */
  isInvalid?: boolean;
  /**
   * If the radio is checked.
   */
  checked?: boolean;
  /**
   * Name of the radio element.
   */
  name?: string;
} & HTMLAttributes<HTMLInputElement>;

/**
 * Base radio element
 */
export function BaseRadio({ isInvalid = false, ...rest }: BaseRadioProps) {
  const [isInvalidState, setIsInvalidState] = useState(isInvalid);

  const handleOnChange = () => {
    setIsInvalidState(false);
  };

  return (
    <Root
      type="radio"
      $isInvalid={isInvalidState}
      onChange={() => handleOnChange()}
      data-testid={BaseRadio.name}
      {...rest}
    />
  );
}
