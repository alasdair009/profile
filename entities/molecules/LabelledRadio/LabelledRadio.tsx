import { Root } from "./styles";
import { BaseRadio } from "@/entities/atoms/BaseRadio";
import { BaseLabel } from "@/entities";
import { BaseRadioProps } from "@/entities/atoms/BaseRadio/BaseRadio";

type LabelledInputProps = {
  /**
   * Text for the label.
   */
  label: string;
} & BaseRadioProps;

/**
 * Labelled text input field for forms.
 */
export function LabelledRadio({ isInvalid, name, label, ...rest }: LabelledInputProps) {
  return (
    <Root data-testid={LabelledRadio.name}>
      <BaseLabel as="span">{label}</BaseLabel>
      <BaseRadio isInvalid={isInvalid} {...rest} />
    </Root>
  );
}
