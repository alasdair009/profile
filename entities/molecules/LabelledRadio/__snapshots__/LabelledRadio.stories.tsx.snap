// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`molecules/LabelledRadio Checked play-test 1`] = `
<div data-testid="LabelledRadio"
     class="sc-braxZu ldDAHa"
>
  <span data-testid="BaseLabel"
        class="sc-elDIKY bvcFdL"
  >
    Radio entry
  </span>
  <input data-testid="BaseRadio"
         required
         class="sc-gJhJTp hddwbN"
         type="radio"
         checked
  >
</div>
`;

exports[`molecules/LabelledRadio Default play-test 1`] = `
<div data-testid="LabelledRadio"
     class="sc-braxZu ldDAHa"
>
  <span data-testid="BaseLabel"
        class="sc-elDIKY bvcFdL"
  >
    Radio entry
  </span>
  <input data-testid="BaseRadio"
         required
         class="sc-gJhJTp hddwbN"
         type="radio"
  >
</div>
`;
